From 4e8911f293e4cc7cf6f312f2b919b4a8cf270450 Mon Sep 17 00:00:00 2001
From: Thomas Colliers <mail@thomascolliers.com>
Date: Sun, 16 Aug 2026 20:30:35 +0200
Subject: [PATCH 5/7] dt-bindings: media: i2c: add OmniVision OV13B10

Document the OV13B10 image sensor, its clock and supplies, optional
reset line, and two- or four-lane CSI-2 interface.

Assisted-by: pi:claude-opus-5
Signed-off-by: Thomas Colliers <mail@thomascolliers.com>
---
 .../bindings/media/i2c/ovti,ov13b10.yaml      | 112 ++++++++++++++++++
 MAINTAINERS                                   |   1 +
 2 files changed, 113 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/media/i2c/ovti,ov13b10.yaml

diff --git a/Documentation/devicetree/bindings/media/i2c/ovti,ov13b10.yaml b/Documentation/devicetree/bindings/media/i2c/ovti,ov13b10.yaml
new file mode 100644
index 000000000000..4073b822621e
--- /dev/null
+++ b/Documentation/devicetree/bindings/media/i2c/ovti,ov13b10.yaml
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/media/i2c/ovti,ov13b10.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: OmniVision OV13B10 CMOS Sensor
+
+maintainers:
+  - Arec Kao <arec.kao@intel.com>
+
+description:
+  The OmniVision OV13B10 is a 13-megapixel CMOS image sensor with a
+  4208x3120 active pixel array. Image data is transmitted through a MIPI
+  CSI-2 interface with either two or four data lanes.
+
+allOf:
+  - $ref: /schemas/media/video-interface-devices.yaml#
+
+properties:
+  compatible:
+    const: ovti,ov13b10
+
+  reg:
+    maxItems: 1
+
+  clocks:
+    description: 19.2 MHz external clock
+    maxItems: 1
+
+  avdd-supply:
+    description: Analog power supply
+
+  dovdd-supply:
+    description: Digital I/O power supply
+
+  dvdd-supply:
+    description: Digital core power supply
+
+  reset-gpios:
+    description: Active-low sensor reset
+    maxItems: 1
+
+  port:
+    $ref: /schemas/graph.yaml#/$defs/port-base
+    additionalProperties: false
+
+    properties:
+      endpoint:
+        $ref: /schemas/media/video-interfaces.yaml#
+        unevaluatedProperties: false
+
+        properties:
+          data-lanes:
+            oneOf:
+              - minItems: 2
+                maxItems: 2
+              - minItems: 4
+                maxItems: 4
+
+          link-frequencies:
+            items:
+              - const: 560000000
+
+        required:
+          - data-lanes
+          - link-frequencies
+
+    required:
+      - endpoint
+
+required:
+  - compatible
+  - reg
+  - clocks
+  - avdd-supply
+  - dovdd-supply
+  - dvdd-supply
+  - port
+
+unevaluatedProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/gpio/gpio.h>
+
+    i2c {
+        #address-cells = <1>;
+        #size-cells = <0>;
+
+        camera@36 {
+            compatible = "ovti,ov13b10";
+            reg = <0x36>;
+
+            clocks = <&camera_clk>;
+
+            avdd-supply = <&camera_avdd_2v8>;
+            dovdd-supply = <&camera_dovdd_1v8>;
+            dvdd-supply = <&camera_dvdd_1v1>;
+
+            reset-gpios = <&gpio 41 GPIO_ACTIVE_LOW>;
+
+            port {
+                ov13b10_out: endpoint {
+                    remote-endpoint = <&csi_in>;
+                    data-lanes = <0 1 2 3>;
+                    link-frequencies = /bits/ 64 <560000000>;
+                };
+            };
+        };
+    };
+...
diff --git a/MAINTAINERS b/MAINTAINERS
index 8014b9f8253e..bb669c668272 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -19963,6 +19963,7 @@ M:	Arec Kao <arec.kao@intel.com>
 L:	linux-media@vger.kernel.org
 S:	Maintained
 T:	git git://linuxtv.org/media.git
+F:	Documentation/devicetree/bindings/media/i2c/ovti,ov13b10.yaml
 F:	drivers/media/i2c/ov13b10.c
 
 OMNIVISION OV2680 SENSOR DRIVER
-- 
2.55.0

